commands/next_lint_error.py

```diff
@@ -57,25 +57,17 @@ def _harvest_next(self) -> int:
             self.window.active_view().sel()[0].begin()
         )
         lines = set([])
-        next_change = None
         vid = self.window.active_view().id()
         for error_type in ['ERRORS', 'WARNINGS', 'VIOLATIONS']:
             for line, _ in ANACONDA[error_type].get(vid, {}).items():
                 lines.add(int(line))
 
-        lines = list(set(sorted(lines)))
+        lines = set(sorted(lines))
         if not len(lines):
             return None
 
-        if not cur_line:
-            return None
-
-        if lines[-1] > cur_line:
-            for l in lines:
-                if l > cur_line:
-                    next_change = l
-                    break
-        else:
-            next_change = lines[0]
+        lines_list = []
+        if cur_line and list(lines)[-1] > cur_line:
+            lines_list = [l for l in lines if l > cur_line]
 
-        return next_change
+        return lines_list[0] if len(lines_list) > 0 else lines.pop()
```

Comment:
Yeah simpler and confusing as well.

I normally write a Python module for anaconda and then two years after that I have to come back to add new functionality or to fix a bug, in all that time I don't open the file (if is not part of the core) even once so when I come back I like to be able to read the code and totally understand what is doing.

That is why I always follow the Python zen "explicit is better than implicit" and the mantra "readability matters", I dislike "implicit boolean" conversions and I try to avoid them not only in Python but in any language that I write code in.

If we use something like `return lines_list[0] if lines_list else lines.pop()` we are making and implicit boolean conversion for a type that should never be interpreted as a boolean, that is why the explicit check `len(lines_list) > 0` give us information about what the type is and what we meant with the check. What we want to check is if the variable has no elements, that is quite different to check if the variable is `True`.

Here and as I also try to use `meaningful variable names` it doesn't matter too much cos there is `list` word in the name of the variable, but, in any case, I prefer to be explicit (explicit is not the same as verbose) and the whole anaconda code is written with those concepts in mind.